### PR TITLE
Rework Order to use Ordering and test instance laws

### PR DESCRIPTION
**Pull request by 1Jajen1**
Two changes worth noting:
- Orders main combinator now returns `Ordering`. This has a few advantages: It is much easier to reason about than integers and second it offers a monoid instance so we can combine the result of `compare` to obtain a new `Ordering`. This allows nice and easy definitions for types with multiple values.
- Added laws and tests for the `Order` typeclass. It was previously lawless.

### CODE REVIEW

File: arrow-core/src/main/kotlin/arrow/core/extensions/boolean.kt [hunk 1]
```diff
@@ -14,6 +16,11 @@ interface BooleanEq : Eq<Boolean> {
   override fun Boolean.eqv(b: Boolean): Boolean = this == b
 }
 
+interface BooleanOrder : Order<Boolean> {
+  override fun Boolean.compare(b: Boolean): Ordering = Ordering.fromInt(this.compareTo(b))
```
Comment (nomisRev) [inline arrow-core/src/main/kotlin/arrow/core/extensions/boolean.kt:20]: Nit: impls on new lines

**Comment by nomisRev** [comment]
Thanks @1Jajen1! I think this was one of my early contributions to Arrow 😅
Using `Int` over `Ordering` was definitely the wrong call.

**Comment by 1Jajen1** [comment]
The build fails with a bug in kapt by the looks of it. Locally compiles fine. This looks weird as if it has a cache issue because it seems related to `Eq` being a superclass of `Order` which was removed in this pr

**Comment by 1Jajen1** [comment]
> The build fails with a bug in kapt by the looks of it. Locally compiles fine.

Got this fixed. My guess is that github actions run on a machine that considers the paths: "x/EQ.java" and "x/Eq.java" to be the same whereas my machine does not. Only way I can explain why this failed in actions only :sweat_smile: 

Either way I moved the `Ordering` to the datatypes where it belonged in the first place.

**Comment by 1Jajen1** [comment]
> arrow.fx.QueueTest > BoundedQueue - peekAll reads all values, including outstanding offers FAILED

I guess that is a flaky test?

**Comment by nomisRev** [comment]
@1Jajen1 yes, it's flaky due to CI imprecision on timing :/

**Comment by rachelcarmena** [comment]
> My guess is that github actions run on a machine that considers the paths: "x/EQ.java" and "x/Eq.java" to be the same whereas my machine does not. 

Right @1Jajen1 , it runs on a macOS environment with case insensitive filesystem :raised_hands:

**Pull request merged**